pkg/kubectl/stop.go

```diff
@@ -276,6 +283,69 @@ func (reaper *JobReaper) Stop(namespace, name string, timeout time.Duration, gra
 	return nil
 }
 
+func (reaper *DeploymentReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *api.DeleteOptions) error {
+	deployments := reaper.Extensions().Deployments(namespace)
+	replicationControllers := reaper.ReplicationControllers(namespace)
+	scaler, err := ScalerFor(extensions.Kind("Deployment"), *reaper)
+	if err != nil {
+		return err
+	}
+	deployment, err := deployments.Get(name)
+	if err != nil {
+		return err
+	}
+
+	// clear deployment history
+	zero := 0
+	deployment.Spec.RevisionHistoryLimit = &zero
+	deployment, err = deployments.Update(deployment)
```

Comment:
Why not update both d.spec.replicas and d.spec.revisionHistoryLimit? Then you won't need to scale down since you already poll d.status.replicas